Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/EditorWidgetAction.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/std/containers/unordered_set.h>
+#include <AzCore/std/function/function_template.h>
+#include <AzCore/std/string/string.h>
+
+#include <QAction>
+#include <QIcon>
+
+namespace AzToolsFramework
+{
+    class EditorWidgetAction
```

Comment:
suggestion: API comment here would definitely be useful

Reply:
These are internal classes so it's not as important, but I'll try to add relevant implementation comments in follow-up PRs.